llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -5495,6 +5495,34 @@ bool AMDGPULegalizerInfo::legalizeLDSKernelId(MachineInstr &MI,
   return true;
 }
 
+bool AMDGPULegalizerInfo::legalizePreloadImplicitarg(
+    MachineInstr &MI, MachineRegisterInfo &MRI, MachineIRBuilder &B) const {
+  assert(ST.hasKernargPreload());
```

Comment:
This shouldn't be an assert. Better to just return false and let the intrinsic fail to select